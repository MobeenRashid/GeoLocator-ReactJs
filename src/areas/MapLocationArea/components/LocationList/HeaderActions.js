import React from 'react';
import { Button } from 'antd';
import { Link } from 'react-router-dom';

const HeaderActions = (props) => {
    return (
        <Button type="primary" onClick={props.onAddMapClick}>
            <Link to="locations/addnew">Add Map</Link>
        </Button>
    );
}

export default HeaderActions;