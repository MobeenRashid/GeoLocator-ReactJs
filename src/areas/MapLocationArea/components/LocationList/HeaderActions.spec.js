import React from 'react';
import { render, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import HeaderActions from './HeaderActions';

describe('HeaderActions component', () => {

    test("Should render correctly", () => {
        const component = render(<BrowserRouter>
            <HeaderActions></HeaderActions>
        </BrowserRouter>);
        expect(component).toMatchSnapshot();
    });

    test("Should call onAddMapClick", () => {
        const clickFn = jest.fn();
        const component = mount(<BrowserRouter>
            <HeaderActions onAddMapClick={clickFn}></HeaderActions>
        </BrowserRouter>);

        component.find('#btnAddMap').first().simulate('click');
        expect(clickFn).toBeCalled();
        component.unmount();
    });
});