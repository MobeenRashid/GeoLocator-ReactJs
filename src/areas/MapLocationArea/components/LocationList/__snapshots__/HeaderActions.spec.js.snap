// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderActions component Should render correctly 1`] = `
<button
  class="ant-btn ant-btn-primary"
  id="btnAddMap"
  type="button"
>
  <a
    href="/locations/addnew"
  >
    Add Map
  </a>
</button>
`;
